include/bluetooth/audio/has.h

```diff
@@ -0,0 +1,350 @@
+ */
+int bt_has_discover(struct bt_conn *conn);
+
+/** @brief Set Preset Name.
+ *
+ *  Used by client and server to set the preset name identified by the index.
+ *  Once used by the server to change the name of local preset, @ref bt_has_ops->name_changed
+ *  callback will be called.
+ *
+ *  @note For server, this API is only available when the
+ *        @kconfig{CONFIG_BT_HAS_PRESET_NAME_DYNAMIC} configuration option has been enabled.
+ *
+ *  @param has Pointer to the Hearing Access Service object.
+ *  @param index Preset index.
+ *  @param name Preset name to be written.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_has_preset_name_set(struct bt_has *has, uint8_t index, const char *name);
+
+/** @brief Get Active Preset.
+ *
+ *  Used by client to get the index of currently active preset on remote server.
+ *
+ *  The value is returned in the bt_has_client_cb.active_preset callback.
+ *
+ *  @param has Pointer to the Hearing Access Service object.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_has_preset_active_get(struct bt_has *has);
+
+/** @brief Set Active Preset.
+ *
+ *  Used by client and server to set the preset identified by the ID as the active preset.
+ *
+ *  @param has Pointer to the Hearing Access Service object.
+ *  @param index Preset index.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_has_preset_active_set(struct bt_has *has, uint8_t index);
+
+/** @brief Set Next Preset.
+ *
+ *  Used by client to set the next preset record on the server list as the active preset.
+ *
+ *  @param has Pointer to the Hearing Access Service object.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_has_preset_active_set_next(struct bt_has *has);
+
+/** @brief Set Previous Preset.
+ *
+ *  Used by client to set the previous preset record on the server list as the active preset.
+ *
+ *  @param has Pointer to the Hearing Access Service object.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_has_preset_active_set_prev(struct bt_has *has);
+
+/** @brief Clear out Active Preset.
+ *
+ *  Used by server to deactivate currently active preset.
+ *
+ *  @param has Pointer to the Hearing Access Service object.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_has_preset_active_clear(struct bt_has *has);
+
+/** @brief Set visibility of preset record.
+ *
+ *  Server method to set visibility of preset record. Invisible presets are not removed
+ *  but are not visible by peer devices.
+ *
+ *  @param has Pointer to the Hearing Access Service object.
+ *  @param index Preset index.
+ *  @param visible true if preset should be visible
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_has_preset_visibility_set(struct bt_has *has, uint8_t index, bool visible);
```

Comment:
What is the purpose and reason for (adding) this concept of "visibility" for presets?  It does not seem to be a feature of the specification.